solutions/cses-1158.mdx

```diff
@@ -14,7 +14,47 @@ author: Sofia Yang
 ## Implementation
 
 <LanguageSection>
+	
+<CPPSection>
 
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	int n, x;
+	cin >> n >> x;
+	vector<int> cost(n + 1);
+	vector<int> pages(n + 1);
+
+	for (int i = 1; i <= n; i++) {
+		cin >> cost[i];
+	}
+	for (int i = 1; i <= n; i++) {
+		cin >> pages[i];
+	}
```

Comment:
please make the arrays 0-indexed

Reply:
won't it be easier to follow along if the dp and array indices match? the dp indices have to be 1 indexed because the 0th index is for the base case.

Reply:
> won't it be easier to follow along

no, not really?